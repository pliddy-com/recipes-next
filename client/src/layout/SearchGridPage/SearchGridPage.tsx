import Container from '@mui/material/Container';

import SearchResults from 'components/Search/SearchResults/SearchResults';
import SearchBox from 'components/Search/SearchBox/SearchBox';
import Typography from '@mui/material/Typography';

interface SearchGridPageProps {
  title?: string | null;
}
const SearchGridPage = ({ title }: SearchGridPageProps) => {
  return (
    <Container
      className="page recipegrid search"
      data-testid="page"
      maxWidth="xl"
    >
      <Typography variant="h1" sx={{ marginBottom: '0.75rem' }}>
        {title}
      </Typography>

      <SearchBox />
      <SearchResults />
    </Container>
  );
};

export default SearchGridPage;
