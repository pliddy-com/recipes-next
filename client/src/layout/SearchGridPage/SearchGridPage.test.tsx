import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import SearchGridPage from './SearchGridPage';

jest.mock('components/Search/SearchBox/SearchBox');
jest.mock('components/Search/SearchResults/SearchResults');

describe('SearchGridPage', () => {
  // const env = process.env;

  describe('when there is page content', () => {
    it('it renders the SearchGridPage', async () => {
      const title = 'Title';

      const { asFragment, queryByTestId } = render(
        <SearchGridPage title={title} />
      );

      // assert that content is rendered
      expect(queryByTestId('page')).toBeInTheDocument();

      // assert that the component matches the existing snapshot
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
