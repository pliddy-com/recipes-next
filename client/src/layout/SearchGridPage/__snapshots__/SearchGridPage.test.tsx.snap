// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchGridPage when there is page content it renders the SearchGridPage 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl page recipegrid search css-19r6kue-MuiContainer-root"
    data-testid="page"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-1nu87ig-MuiTypography-root"
    >
      Title
    </h1>
    <div
      data-testid="search-box"
    >
      SearchBox
    </div>
    <div
      data-testid="search-results"
    >
      SearchResults
    </div>
  </div>
</DocumentFragment>
`;
